scapy/utils.py

```diff
@@ -1305,6 +1306,15 @@ def write(self, pkt):
             else:
                 pkt = pkt.__iter__()
             for p in pkt:
+
+                if self.previous_packet_type and \
```

Comment:
I'd rather have such a warning written only when the new packet type would result in a different `linktype` value. Some `linktype` values actually **do** accept different layer types (e.g., `Dot3` and `Ether` or `IP` and `IPv6`).

Reply:
Thanks for the idea. That makes more sense. I just pushed a modified version of the PR.